Sources/SwiftCentrifuge/Client.swift

```diff
@@ -450,121 +450,53 @@ fileprivate extension CentrifugeClient {
         self.conn?.write(data: data)
     }
     
-    func waitForReply(id: UInt32, completion: @escaping (Proto_Reply?, Error?)->()) {
-        let group = DispatchGroup()
-        group.enter()
-        
-        var reply: Proto_Reply = Proto_Reply()
-        var groupLeft = false
-        var timedOut = false
-        var opError: Error? = nil
-        
-        let timeoutTask = DispatchWorkItem {
-            if groupLeft {
-                return
-            }
-            timedOut = true
-            groupLeft = true
-            group.leave()
-        }
-        
-        func resolve(rep: CentrifugeResolveData) {
-            if groupLeft {
-                return
-            }
-            if let err = rep.error {
-                opError = err
-            } else if let r = rep.reply {
-                reply = r
-            }
-            timeoutTask.cancel()
-            groupLeft = true
-            group.leave()
+    private func waitForReply(id: UInt32, completion: @escaping (Proto_Reply?, Error?)->()) {
+        let timeoutTask = DispatchWorkItem { [weak self] in
+            self?.opCallbacks[id] = nil
+            completion(nil, CentrifugeError.timeout)
         }
-        let resolveFunc = resolve
-        self.opCallbacks[id] = resolveFunc
         self.syncQueue.asyncAfter(deadline: .now() + self.config.timeout, execute: timeoutTask)
         
-        // wait for resolve or timeout
-        self.workQueue.async { [weak self] in
-            guard let strongSelf = self else { return }
-            group.wait()
-            strongSelf.syncQueue.async { [weak self] in
-                guard let strongSelf = self else { return }
-                strongSelf.opCallbacks.removeValue(forKey: id)
-                if let err = opError {
-                    completion(nil, err)
-                    return
-                }
-                if timedOut {
-                    completion(nil, CentrifugeError.timeout)
-                    return
-                }
-                completion(reply, nil)
+        self.opCallbacks[id] = { [weak self] rep in
+            timeoutTask.cancel()
+            
+            self?.opCallbacks[id] = nil
```

Comment:
I just understood that remove from dictionary is better something into the 'handleData' function

Reply:
After deep researching I understood that in this case we must remove callback from dictionary by the hands. But in others same cases It's unnecessary.